c7n/resources/glacier.py

```diff
@@ -93,3 +96,70 @@ def _augment(r):
 
         return super(GlacierCrossAccountAccessFilter, self).process(
             resources, event)
+
+
+@Glacier.action_registry.register('remove-statements')
+class RemovePolicyStatement(RemovePolicyBase):
+    """Action to remove policy statements from Glacier
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: glacier-cross-account
+                resource: glaier
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: remove-statements
+                    statement_ids: matched
+    """
+
+    permissions = ('glacier:SetVaultAccessPolicy', 'glacier:GetVaultAccessPolicy')
+
+    def process(self, resources):
+        results = []
+        client = local_session(self.manager.session_factory).client('glacier')
+        for r in resources:
+            try:
+                if self.process_resource(client, r):
+                    results.append(r)
+            except:
+                self.log.exception(
+                    "Error processing glacier:%s", r['VaultARN'])
+        return results
+
+    def process_resource(self, client, resource):
+        if 'Policy' not in resource:
+            try:
+                resource['Policy'] = client.get_vault_access_policy(
+                    vaultName=resource['VaultName'])['policy']['Policy']
+            except ClientError as e:
+                if e.response['Error']['Code'] != "ResourceNotFoundException":
+                    raise
+                resource['Policy'] = None
+
+        if not resource['Policy']:
+            return
+
+        p = json.loads(resource['Policy'])
+        statements, found = self.process_policy(
+            p, resource, CrossAccountAccessFilter.annotation_key)
+
+        if not found:
+            return
+
+        if not statements:
+            client.delete_vault_access_policy(
+                vaultName=resource['VaultName'])
+        else:
+            client.set_vault_access_policy(
```

Comment:
this common code path for modifying the vault policy is never tested.

Reply:
resolved in latest changes to tests